okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java

```diff
@@ -208,18 +217,23 @@ private void connectTunnel(int connectTimeout, int readTimeout, int writeTimeout
       rawSocket = null;
       sink = null;
       source = null;
+      eventListener.connectEnd(call, route.socketAddress(), null, null);
```

Comment:
This is an oddball case: we attempted to create a tunnel, and forced to authenticate. The server closed the connection so we must reconnect with the authentication details. The connection itself is perfectly fine, but we can't derive the protocol at this point.